pyomo/solvers/plugins/solvers/mosek_direct.py

```diff
@@ -163,21 +163,25 @@ def _process_stream(msg):
         for key, option in self.options.items():
             try:
                 param = key.split('.')
-                if param[0] == 'mosek':
-                    param.pop(0)
-                param = getattr(mosek, param[0])(param[1])
-                if 'sparam' in key.split('.'):
-                    self._solver_model.putstrparam(param, option)
-                elif 'dparam' in key.split('.'):
-                    self._solver_model.putdouparam(param, option)
-                elif 'iparam' in key.split('.'):
-                    if isinstance(option, str):
-                        option = option.split('.')
-                        if option[0] == 'mosek':
-                            option.pop('mosek')
-                        option = getattr(mosek, option[0])(option[1])
-                    else:
-                        self._solver_model.putintparam(param, option)
+                if key == param[0]:
+                    self._solver_model.putparam(key, option)
+                else:
+                    if param[0] == 'mosek':
+                        param.pop(0)
+                    param = getattr(mosek, param[0])(param[1])
+                    if 'sparam' in key.split('.'):
+                        self._solver_model.putstrparam(param, option)
+                    elif 'dparam' in key.split('.'):
+                        self._solver_model.putdouparam(param, option)
+                    elif 'iparam' in key.split('.'):
+                        if isinstance(option, int):
+                            self._solver_model.putintparam(param, option)
+                        elif isinstance(option, str):
```

Comment:
Same here - without an `else`, option values that aren't int / str will be silently skipped (this would include, e.g., numpy ints)

Reply:
I have included a warning under an else statement in this situation.